python/tvm/contrib/nvcc.py

```diff
@@ -269,15 +270,24 @@ def have_int8(compute_version):
     return False
 
 
-def have_tensorcore(compute_version):
+def have_tensorcore(compute_version=None):
     """Either TensorCore support is provided in the compute capability or not
 
     Parameters
     ----------
     compute_version : str
         compute capability of a GPU (e.g. "7.0")
     """
+    if compute_version is None:
+        if tvm.gpu(0).exist:
+            compute_version = tvm.gpu(0).compute_version
+        else:
+            compute_version = AutotvmGlobalScope.current.cuda_target_arch
```

Comment:
- It seems to me that we should move this config to PassContext, because this affects how Relay op strategy select the implementation in general but not specific to AutoTVM.
- Should properly handle the case that `cuda_target_arch` is `None`. Like we could print a warning saying that we will not consider tensor core due to missing information.

Reply:
Those are both good points, thanks @comaniac. I'll take a look at PassContext and see if it can be integrated there instead.

Reply:
Just to be clear, would you want to set 'cuda_target_arch' as part of a `PassContext` config or is there a different approach you were thinking of? When I investigated it wasn't quite as clear a fit as I thought.

Reply:
I was expecting something like:

```python
with tvm.transform.PassContext(opt_level=3, config={"relay.backend.cuda_target_arch": "sm_75"}):
    lib = relay.build(mod, target=target, params=params)
```

And in `ncvv.py`:

```python
cuda_target_arch = PassContext.current().config.get("relay.backend.cuda_target_arch", "sm_75")
major, minor = cuda_target_arch.split("_")[1]
compute_version = major + "." + minor
```

Does that make sense to you?
If so, you may refer to a recent PR I sent, which adds "relay.backend.use_auto_scheduler" to PassContext: https://github.com/apache/tvm/pull/6903

Reply:
I thought about this some more overnight and I'm not sure adding it PassContext makes any more sense than having it in AutoTVMGlobalScope. We really should be specifying this information as part of a `tvm.Target` since the cuda architecture purely describes the hardware target and doesn't really relate to relay passes or autotvm directly. I think this should be done once we move further from string based targets to objects like those introduced in #6218. I'd argue that for now, applying a bandaid fix to Autotvm.GlobalScope as in the current PR is the best way to temporarily solve the problem. What do you think @comaniac? I'd also be interested in hearing what @tqchen thinks.

Reply:
Hmm this is a good point. Putting CUDA target arch to target definitely makes more sense. Then the solution becomes:

```python
target = Target("cuda -arch=sm_80")
...
cuda_target_arch = target.attrs["arch"] if "arch" in target.attrs else "sm_75"
...
```

I found that CUDA target already has this attribute so the above solution actually works now.

Reply:
I've changed `have_tensorcore` to instead extract the architecture from a target. Can you take another look and let me know what you think? One downside is that `have_tensorcore` is used in places that don't have access to the target object such as in topi functions, so we have to support both the old way of getting target from `tvm.gpu...` and the target object.